mantidproject/mantid
# Fix for sequential and Serial fit buttons in Eng Diff interface

**Description of work**

Fix bug highlighted in manual testing where clicking the sequential or serial fit buttons in the Engineering Diffraction interface fitting tab would result in an unhandled error if data was not loaded or if a fitting model was not set.

The buttons are now disabled until it is possible to fit (there is data loaded and there is a model).


**To test:**

- Open the Engineering Diffraction interfaces and go to the fitting tab
- [ ] With no data loaded, the `Sequential Fit` and `Serial Fit` buttons should be disabled
- Follow the steps [here](https://developer.mantidproject.org/Testing/EngineeringDiffraction/EngineeringDiffractionTestGuide.html#focus) to get some focused data (skip if you have some)
- Load that data into the fit tab with `Add to plot` checked
- [ ] Once again, click on the `Sequential Fit` and `Serial Fit` buttons should be disabled since there is no model set
- Click the `Fit` button to bring up the fit property browser interface
- Right click and add a peak to the plot (so there is now a model)
- [ ] `Sequential Fit` and `Serial Fit` should now both run when their corresponding buttons are clicked

Test other scenarios that you can think of if you are more familiar with this interface than me

Fixes #35595

*This does not require release notes* because **it's a regression fixed before release**

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
